crates/uv-virtualenv/src/virtualenv.rs

```diff
@@ -57,32 +57,26 @@ pub(crate) fn create(
     // considered the "base" for the virtual environment. This is typically the Python executable
     // from the [`Interpreter`]; however, if the interpreter is a virtual environment itself, then
     // the base Python executable is the Python executable of the interpreter's base interpreter.
-    let base_python = if cfg!(unix) {
-        // On Unix, follow symlinks to resolve the base interpreter, since the Python executable in
-        // a virtual environment is a symlink to the base interpreter.
-        uv_fs::canonicalize_executable(interpreter.sys_executable())?
-    } else if cfg!(windows) {
-        // On Windows, follow `virtualenv`. If we're in a virtual environment, use
-        // `sys._base_executable` if it exists; if not, use `sys.base_prefix`. For example, with
-        // Python installed from the Windows Store, `sys.base_prefix` is slightly "incorrect".
-        //
-        // If we're _not_ in a virtual environment, use the interpreter's executable, since it's
-        // already a "system Python". We canonicalize the path to ensure that it's real and
-        // consistent, though we don't expect any symlinks on Windows.
-        if interpreter.is_virtualenv() {
-            if let Some(base_executable) = interpreter.sys_base_executable() {
-                base_executable.to_path_buf()
-            } else {
-                // Assume `python.exe`, though the exact executable name is never used (below) on
-                // Windows, only its parent directory.
-                interpreter.sys_base_prefix().join("python.exe")
-            }
-        } else {
-            interpreter.sys_executable().to_path_buf()
-        }
-    } else {
-        unimplemented!("Only Windows and Unix are supported")
-    };
+    let mut base_python = std::path::absolute(
+        interpreter
+            .sys_base_executable()
+            .unwrap_or(interpreter.sys_executable()),
+    )?;
+
+    // If the interpreter isn't in the scripts path, assume it's a symlink. For example, if the user
+    // symlinked a `python-build-standalone` executable to another directory, the `sys._base_executable`
+    // will point to the symlink, not the actual Python executable.
+    while !base_python.starts_with(interpreter.scripts()) {
+        let Some(resolved_python) = uv_fs::read_executable_link(&base_python)? else {
+            break;
+        };
+        debug!(
+            "Resolved symlink from `{}` to `{}`",
+            base_python.simplified_display(),
+            resolved_python.simplified_display()
+        );
+        base_python = resolved_python;
+    }
```

Comment:
This is my attempt to solve the problem demonstrated by our test suite. We symlink from the installed Python directory to (e.g.) `/Users/crmarsh/.local/share/uv/tests/.tmpIl4Pjf/python/3.12/python3`. In `python-build-standalone`, `sys._base_executable` for that symlink just points to the current Python (i.e., it points to `/Users/crmarsh/.local/share/uv/tests/.tmpIl4Pjf/python/3.12/python3`). The trick I'm employing here is that we resolve symlinks as long as the binary is not in the `scripts` path.

Reply:
Does this only occur with python build standalone? In this case, we should patch this in pbs

Reply:
Patch it to what?

Reply:
i think we have to fix https://github.com/indygreg/python-build-standalone/issues/380, or at least i expect it's closely related.

Reply:
What do you think of the solution itself? Alternatively, we could determine whether this is a uv-managed Python, and resolve the symlink if so. (But we might then fail if users install PBS builds through some other mechanism and symlink _those_.)

Reply:
(replied in the main thread)